velox/docs/functions/spark/decimal.rst

```diff
@@ -2,47 +2,92 @@
 Decimal Operators
 =================
 
-When calculating the result precision and scale of arithmetic operators,
-the formulas follow Hive which is based on the SQL standard and MS SQL:
+The result precision and scale computation of arithmetic operators contains two stages.
+In the first stage, the formulas follow Hive which is based on the SQL standard and MS SQL.
+In the second stage, caps p at 38 and reduces scale or not according to the allowing precision loss mode.
+
+Decimal Precision and Scale Computation Formulas
+------------------------------------------------
+
+The SQL standard and MS SQL:
 
 https://cwiki.apache.org/confluence/download/attachments/27362075/Hive_Decimal_Precision_Scale_Support.pdf
 
 https://msdn.microsoft.com/en-us/library/ms190476.aspx
 
+Additionally, the computation of decimal division adapts to the allowing precision loss mode,
+while the decimal addition, subtraction, and multiplication do not.
+
 Addition and Subtraction
-------------------------
+~~~~~~~~~~~~~~~~~~~~~~~~
 
 ::
 
 	p = max(p1 - s1, p2 - s2) + max(s1, s2) + 1
 	s = max(s1, s2)
 
 Multiplication
---------------
+~~~~~~~~~~~~~~
 
 ::
 
 	p = p1 + p2 + 1
 	s = s1 + s2
 
 Division
---------
+~~~~~~~~
+When allowing precision loss:
 
 ::
 
     p = p1 - s1 + s2 + max(6, s1 + p2 + 1)
     s = max(6, s1 + p2 + 1)
 
-For above arithmetic operators, when the precision of result exceeds 38,
-caps p at 38 and reduces the scale, in order to prevent the truncation of
-the integer part of the decimals. Below formula illustrates how the result
-precision and scale are adjusted.
+When denying precision loss:
+
+::
+
+    wholeDigits = min(38, p1 - s1 + s2);
+    fractionalDigits = min(38, max(6, s1 + p2 + 1));
+    p = wholeDigits + fractionalDigits
+    s = fractionalDigits
+
+Decimal Precision and Scale Adjustment
+--------------------------------------
+
+When allowing precision loss, rounding the decimal part of the result if an exact representation is not possible.
+Otherwise, NULL is returned in those cases, as previously.
```

Comment:
> as previously

It is not clear what this refers to here. Perhaps, drop.

> Otherwise, NULL is returned in those cases, as previously.

Otherwise, returns NULL.

> rounding the decimal ...

rounds the decimal ...

Use active verbs for readability.